Update docs and sample-config

abbra opened the pull request:
Tried to configure ipatool from scratch.

abbra committed:
Make sure 'ipatool sample-config' actually works

abbra committed:
Add 'ipatool backport' command

stlaz commented:
Today I tried doing `ipatool backport 1110 --branch ipa-4-6` but that did not work for me.

abbra commented:
Any specific error messages?

stlaz commented:
No, I just issued the command, it even exited with 0:
```
$ ipatool -v backport 1110 --branch ipa-4-6
Options:
{'--': False,
 '--am': False,
 '--backport': [],
 '--branch': ['ipa-4-6'],
 '--color': 'auto',
 '--comment': None,
 '--config': '~/.ipa/toolconf.yaml',
 '--dry-run': False,
 '--force': False,
 '--help': False,
 '--label': [],
 '--no-fetch': False,
 '--no-pagure': False,
 '--no-reviewer': False,
 '--reviewer': [],
 '--state': [],
 '--ticket': [],
 '--verbose': 1,
 'PATCH': [],
 'PR_ID': '1110',
 'all': 0,
 'am': False,
 'backport': True,
 'closed': 0,
 'open': 0,
 'pr-ack': False,
 'pr-list': False,
 'pr-push': False,
 'pr-reject': False,
 'push': False,
 'sample-config': False,
 'start-review': False}
Config:
{'am-command': ['ssh', 'ipa-devel-vm.local', 'cd ~/freeipa/ ; git am -3'],
 'browser': 'firefox',
 'bugzilla-bug-url': 'https://bugzilla.redhat.com/show_bug.cgi?id=',
 'clean-repo-path': '/home/standa/freeipa-pushing',
 'commit-url': 'https://pagure.io/freeipa/c/',
 'gh-fork-remote': 'private',
 'gh-repo': 'freeipa/freeipa',
 'gh-token': '8bbb2d5256c88bc70852639f51ce7412fd77933b',
 'pagure-repository': 'freeipa',
 'pagure-token': 'WXV0O29XROFQXSB2T98Z7B7HLZVCDOPH0CX4255DZGLNA2HV2TI1IJA3KUA0PN7P',
 'patchdir': '/home/standa/freeipa-patches',
 'remote': 'origin',
 'ticket-url': 'https://pagure.io/freeipa/issue/',
 'trac-username-map': {'abbra': 'Alexander Bokovoy <abokovoy@redhat.com>'},
 'username': 'stlaz'}
date -Iseconds
git rev-parse --abbrev-ref HEAD
```

abbra committed:
Update base release version

abbra committed:
Allow to specify git repo via command line

Use --repo to specify path to a clean git repository

abbra committed:
Pick up fixed tickets from commit messages

In addition to picking up fixed issues from milestones, infer them from
the commit messages too. This allows to pick up all fixes for a next
major release.

abbra committed:
Adopt to python3 and use mailmap explicitly
